app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java

```diff
@@ -401,6 +402,21 @@ private boolean notificationSetter(Context context) {
         } else {
             clearAllCalibrationNotifications();
         }
+
+        // Sensor age alert to notify about upcoming expiry
+        long sensorAge = new Date().getTime() - sensor.started_at;
+        if (sensorAge >= prefs.getLong("sensor_age_alerts_days", 13) * 60000 * 24) {
+            if (!prefs.getBoolean("sensor_age_alerts_notified", false)) {
+                OtherAlert(context, "sensor_age_alerts", "Sensor age: " + JoH.niceTimeScalar(sensorAge),
+                        sensorAgeAlertNotificationId, NotificationChannels.SENSOR_AGE_CHANNEL, true,
+                        Integer.MAX_VALUE / 100000);
+                prefs.edit().putBoolean("sensor_age_alerts_notified", true).apply();
+            }
+        } else { // Sensor is still young and fresh
+            notificationDismiss(sensorAgeAlertNotificationId);
+            prefs.edit().putBoolean("sensor_age_alerts_notified", false).apply();
+        }
+
```

Comment:
For state storage use `PersistentStore` instead of `prefs`
Use `Constants.DAY_IN_MS` instead of manually calculating the value for code clarity
There is a msSince() helper method so you don't need to manually calculate date subtractions etc
As you are storing the notification state, don't dismiss the notification unless the state is true to reduce overhead

From a user interface design point of view, I think that really we should be setting a preference of "notify hours before sensor expiry" and then automatically determining when this is. The problem is of course that that information isn't exposed in a unified way at the moment so it wouldn't be implemented easily. This approach can be used for now.

Reply:
I had problem last time that `prefs.getLong("sensor_age_alerts_days", 13)` has crashed xdrip. When debugging the value was stored as string in prefs. Was not investigating further for the reason. @jamorham should it be safe to use getLong() ?

Reply:
I had to solve the crashing so I created this push request https://github.com/danielhers/xDrip/pull/1